### PR TITLE
[NIC-10] Add sign-up and delete person logic

### PR DESCRIPTION
**Pull request by AksyonovaKaterina**

### CODE REVIEW

File: backend/person-service/src/main/kotlin/com/nosql/personservice/PersonServiceApplication.kt [hunk 1]
```diff
@@ -0,0 +1,12 @@
...
+
+import org.springframework.boot.autoconfigure.SpringBootApplication
+import org.springframework.boot.autoconfigure.jdbc.DataSourceAutoConfiguration
+import org.springframework.boot.runApplication
+
+@SpringBootApplication(exclude = [DataSourceAutoConfiguration::class])
+class PersonServiceApplication
+
+fun main(args: Array<String>) {
+	runApplication<PersonServiceApplication>(*args)
```
Comment (V7074) [inline backend/person-service/src/main/kotlin/com/nosql/personservice/PersonServiceApplication.kt]:
```suggestion
	runApplication<PersonServiceApplication>(args = args)
```

File: backend/person-service/src/main/kotlin/com/nosql/personservice/common/exception/ApplicationException.kt [hunk 2]
```diff
@@ -0,0 +1,12 @@
+package com.nosql.personservice.common.exception
+
+import com.nosql.personservice.common.exception.enumerator.ApplicationExceptionCodeEnum
+import com.nosql.personservice.common.exception.enumerator.ApplicationExceptionCodeEnum.UNEXPECTED_ERROR
+import org.springframework.http.HttpStatus
+
+open class ApplicationException(
+    open val status: HttpStatus? = null,
+    val code: ApplicationExceptionCodeEnum = UNEXPECTED_ERROR,
```
Comment (V7074) [inline backend/person-service/src/main/kotlin/com/nosql/personservice/common/exception/ApplicationException.kt]: `error` -> `exception`

File: backend/person-service/src/main/kotlin/com/nosql/personservice/common/exception/handler/ExceptionHandler.kt [hunk 3]
```diff
@@ -0,0 +1,113 @@
...
+    @ExceptionHandler(WebExchangeBindException::class)
+    fun handleException(exception: WebExchangeBindException): ResponseEntity<ApplicationErrorDto>? {
+
+        val errorMessageJoiner = StringJoiner(ERROR_MESSAGE_DELIMITER)
+        val bindingResult = exception.bindingResult
+
+        if (bindingResult.hasGlobalErrors()) {
+
+            val globalErrorsMessages = bindingResult.globalErrors.stream()
+                .map { String.format("%s: %s", it.objectName, it.defaultMessage) }
```
Comment (V7074) [inline backend/person-service/src/main/kotlin/com/nosql/personservice/common/exception/handler/ExceptionHandler.kt]: kotlin string template
copy this class from auth-service

**Review by V7074 — CHANGES_REQUESTED** [review]

File: backend/person-service/src/main/kotlin/com/nosql/personservice/component/impl/DefaultPersonComponent.kt [hunk 4]
```diff
@@ -0,0 +1,72 @@
...
+
+        val operationFailedWithMessage = "$operationDetails: failed with"
+        val errorDetails = if (error is ApplicationException) error.description else "something went wrong"
+
+        return when (error) {
+            is DataIntegrityViolationException -> {
+                DataIntegrityViolationApplicationException()
+            }
+            is ApplicationException -> {
+                error.apply { description = """$operationFailedWithMessage error: $errorDetails""" }
```
Comment (V7074) [inline backend/person-service/src/main/kotlin/com/nosql/personservice/component/impl/DefaultPersonComponent.kt]: `"""`

File: backend/person-service/src/main/kotlin/com/nosql/personservice/controller/PersonController.kt [hunk 5]
```diff
@@ -0,0 +1,43 @@
...
+class PersonController(
+    private val personService: PersonService,
+) {
+
+    @PostMapping(
+        path = [SIGN_UP_URL_PATH],
+        consumes = [MediaType.APPLICATION_JSON_VALUE],
+        produces = [MediaType.APPLICATION_JSON_VALUE],
+    )
+    suspend fun signUp(
```
Comment (V7074) [inline backend/person-service/src/main/kotlin/com/nosql/personservice/controller/PersonController.kt:26]: signUp?

File: backend/person-service/src/main/kotlin/com/nosql/personservice/converter/ConfidentialDtoToConfidentialEntityConverter.kt [hunk 6]
```diff
@@ -0,0 +1,23 @@
...
+
+import com.nosql.personservice.dto.ConfidentialDto
+import com.nosql.personservice.entity.ConfidentialEntity
+import org.bson.types.ObjectId
+import org.springframework.core.convert.converter.Converter
+import org.springframework.stereotype.Component
+
+@Component
+class ConfidentialDtoToConfidentialEntityConverter(
+    private val passportDataDtoToPassportDataEntityConverter: PassportDataDtoToPassportDataEntityConverter,
```
Comment (V7074) [inline backend/person-service/src/main/kotlin/com/nosql/personservice/converter/ConfidentialDtoToConfidentialEntityConverter.kt]: maybe use the conversion service with lazy initialization here?

File: backend/person-service/src/main/kotlin/com/nosql/personservice/converter/PersonDtoToPersonEntityConverter.kt [hunk 7]
```diff
@@ -0,0 +1,34 @@
...
+import com.nosql.personservice.entity.PersonEntity
+import org.bson.types.ObjectId
+import org.springframework.core.convert.converter.Converter
+import org.springframework.stereotype.Component
+
+@Component
+class PersonDtoToPersonEntityConverter(
+    private val jobTimeDtoToJobTimeEntityConverter: JobTimeDtoToJobTimeEntityConverter,
+    private val confidentialDtoToConfidentialEntityConverter: ConfidentialDtoToConfidentialEntityConverter,
+    private val contactsDtoToContactsEntityConverter: ContactsDtoToContactsEntityConverter,
```
Comment (V7074) [inline backend/person-service/src/main/kotlin/com/nosql/personservice/converter/PersonDtoToPersonEntityConverter.kt]: maybe use the conversion service with lazy initialization here?

File: backend/person-service/src/main/kotlin/com/nosql/personservice/dto/PersonDto.kt [hunk 8]
```diff
@@ -0,0 +1,56 @@
...
+import com.nosql.personservice.enumerator.PositionEnum
+import com.nosql.personservice.enumerator.SexEnum
+import com.nosql.personservice.enumerator.StatusEnum
+import java.util.Date
+
+@JsonInclude(JsonInclude.Include.NON_NULL)
+@JsonIgnoreProperties(ignoreUnknown = true)
+class PersonDto (
+
+    @JsonProperty
```
Comment (V7074) [inline backend/person-service/src/main/kotlin/com/nosql/personservice/dto/PersonDto.kt]: ("id")

File: backend/person-service/src/main/kotlin/com/nosql/personservice/entity/ProjectEntity.kt [hunk 9]
```diff
@@ -0,0 +1,16 @@
+package com.nosql.personservice.entity
+
+import org.bson.types.ObjectId
+import org.springframework.data.annotation.Id
+import org.springframework.data.mongodb.core.mapping.Document
+import org.springframework.data.mongodb.core.mapping.Field
+
+@Document("project_document")
```
Comment (V7074) [inline backend/person-service/src/main/kotlin/com/nosql/personservice/entity/ProjectEntity.kt]:
```suggestion
@Document("project")
```

and check other entities

File: backend/person-service/src/main/kotlin/com/nosql/personservice/repository/OfficeRepository.kt [hunk 10]
```diff
@@ -0,0 +1,10 @@
+package com.nosql.personservice.repository
+
+import com.nosql.personservice.entity.OfficeEntity
+import org.bson.types.ObjectId
+import org.springframework.data.mongodb.repository.ReactiveMongoRepository
+import org.springframework.stereotype.Repository
+
+@Repository
+interface OfficeRepository : ReactiveMongoRepository<OfficeEntity, ObjectId> {
```
Comment (V7074) [inline backend/person-service/src/main/kotlin/com/nosql/personservice/repository/OfficeRepository.kt]: delete empty curve brackets 

and check other repositories

File: backend/person-service/src/main/kotlin/com/nosql/personservice/service/impl/DefaultPersonService.kt [hunk 11]
```diff
@@ -0,0 +1,28 @@
...
+import com.nosql.personservice.dto.DefaultApiResponseDto
+import com.nosql.personservice.dto.PersonDto
+import com.nosql.personservice.service.PersonService
+import org.bson.types.ObjectId
+import org.springframework.stereotype.Service
+
+@Service
+class DefaultPersonService(
+    private val personComponent: DefaultPersonComponent,
+    private val personDtoToPersonEntityConverter: PersonDtoToPersonEntityConverter,
```
Comment (V7074) [inline backend/person-service/src/main/kotlin/com/nosql/personservice/service/impl/DefaultPersonService.kt]: use the conversion service and the `ConversionServiceUtil.kt` in `auth-service`

**Review by V7074 — CHANGES_REQUESTED** [review]

File: backend/person-service/src/main/kotlin/com/nosql/personservice/PersonServiceApplication.kt (hunk 1 above)
Comment (AksyonovaKaterina) [reply backend/person-service/src/main/kotlin/com/nosql/personservice/PersonServiceApplication.kt]: fixed

File: backend/person-service/src/main/kotlin/com/nosql/personservice/common/exception/ApplicationException.kt (hunk 2 above)
Comment (AksyonovaKaterina) [reply backend/person-service/src/main/kotlin/com/nosql/personservice/common/exception/ApplicationException.kt]: fixed

File: backend/person-service/src/main/kotlin/com/nosql/personservice/common/exception/handler/ExceptionHandler.kt (hunk 3 above)
Comment (AksyonovaKaterina) [reply backend/person-service/src/main/kotlin/com/nosql/personservice/common/exception/handler/ExceptionHandler.kt]: fixed

File: backend/person-service/src/main/kotlin/com/nosql/personservice/component/impl/DefaultPersonComponent.kt (hunk 4 above)
Comment (AksyonovaKaterina) [reply backend/person-service/src/main/kotlin/com/nosql/personservice/component/impl/DefaultPersonComponent.kt]: fixed

File: backend/person-service/src/main/kotlin/com/nosql/personservice/controller/PersonController.kt (hunk 5 above)
Comment (AksyonovaKaterina) [reply backend/person-service/src/main/kotlin/com/nosql/personservice/controller/PersonController.kt:26]: Yes, it is request to add person info while registration

File: backend/person-service/src/main/kotlin/com/nosql/personservice/repository/OfficeRepository.kt (hunk 10 above)
Comment (AksyonovaKaterina) [reply backend/person-service/src/main/kotlin/com/nosql/personservice/repository/OfficeRepository.kt]: fixed

File: backend/person-service/src/main/kotlin/com/nosql/personservice/entity/ProjectEntity.kt (hunk 9 above)
Comment (AksyonovaKaterina) [reply backend/person-service/src/main/kotlin/com/nosql/personservice/entity/ProjectEntity.kt]: fixed

File: backend/person-service/src/main/kotlin/com/nosql/personservice/dto/PersonDto.kt (hunk 8 above)
Comment (AksyonovaKaterina) [reply backend/person-service/src/main/kotlin/com/nosql/personservice/dto/PersonDto.kt]: fixed

File: backend/person-service/src/main/kotlin/com/nosql/personservice/service/impl/DefaultPersonService.kt (hunk 11 above)
Comment (AksyonovaKaterina) [reply backend/person-service/src/main/kotlin/com/nosql/personservice/service/impl/DefaultPersonService.kt]: Fixed, but we can have troubles)))))

File: backend/person-service/src/main/kotlin/com/nosql/personservice/converter/PersonDtoToPersonEntityConverter.kt (hunk 7 above)
Comment (AksyonovaKaterina) [reply backend/person-service/src/main/kotlin/com/nosql/personservice/converter/PersonDtoToPersonEntityConverter.kt]: Fixed, but we can have troubles)))))

File: backend/person-service/src/main/kotlin/com/nosql/personservice/converter/ConfidentialDtoToConfidentialEntityConverter.kt (hunk 6 above)
Comment (AksyonovaKaterina) [reply backend/person-service/src/main/kotlin/com/nosql/personservice/converter/ConfidentialDtoToConfidentialEntityConverter.kt]: Fixed, but we can have troubles)))))

**Review by V7074 — APPROVED** [review]

**Pull request merged**